core/src/commonMain/kotlin/com/powersync/PowerSyncException.kt

```diff
@@ -4,3 +4,16 @@ public class PowerSyncException(
     message: String,
     cause: Throwable,
 ) : Exception(message, cause)
+
+
+/**
+ * Helper class designed to bridge SKIEE methods and allow them to throw
+ * `PowerSyncException`. This is necessary because these exceptions cannot
+ * be thrown directly in Swift.
+ *
+ * The class provides a mechanism to handle exceptions in a way that is
+ * compatible with the Swift environment, ensuring proper error propagation
+ * and handling.
+ */
+@Throws(PowerSyncException::class)
+public fun throwPowerSyncException(exception: PowerSyncException): Unit = throw exception
```

Comment:
Since this will only be used for Swift and nowhere else, can we move this into the `PowerSyncKotlin` project? My understanding is that sources from there would also be considered for SKIEE (but maybe do a local build to validate, if it doesn't work we can keep it here).

Reply:
That is a very good suggestion! It looks like we can move this to `PowerSyncKotlin`. I did a clean build and was able to use this method in the Swift SDK 🙌